ansible/roles/dns-dhcp/tasks/main.yml

```diff
@@ -9,10 +9,17 @@
 #  the blacklisting isn't applied immediately)
 - name: Blacklist 127.0.0.1 as a nameserver for the system resolver
   lineinfile:
-    dest: /etc/resolvconf.conf
+    dest: /etc/resolv.conf
     line: name_server_blacklist=127.0.0.1
     state: present
 
+# Set up ramdisk for later storage of dhcp lease file
+- name: Makedir ramdisk
+  file: path=/mnt/ramdisk state=directory
+
+- name: Mount ramdisk
+  mount: name=/mnt/ramdisk src='tmpfs' fstype=tmpfs opts='size=10m' state=mounted
```

Comment:
Cool. I wrote before coffee had kicked in a gave you bad advice (well, unidiomatic advice, at best). It should be sufficient to use a [handler](http://docs.ansible.com/ansible/playbooks_intro.html) to restart dnsmasq (see "Handlers: Running Operations On Change") on the Mount ramdisk task.